manager.go

```diff
@@ -195,32 +199,39 @@ func (m *Manager) SubsystemUpdates(ctx context.Context) {
 		if err != nil {
 			m.logger.Warnw("running install of new agent version", "error", err)
 		}
-		restartCmd := "using 'systemctl restart viam-agent'"
-		if runtime.GOOS == "windows" {
-			restartCmd = "the viam-agent windows service"
-		}
-		m.logger.Infof("viam-agent update complete, please restart %s", restartCmd)
+	}
+
+	if needRestart {
+		m.viamAgentNeedsRestart = true
 	}
 
 	// Viam Server
 	if m.cfg.AdvancedSettings.DisableViamServer {
 		if err := m.viamServer.Stop(ctx); err != nil {
 			m.logger.Warn(err)
 		}
+		if m.viamAgentNeedsRestart {
+			m.Exit()
+			return
+		}
 	} else {
 		needRestart, err := m.cache.UpdateBinary(ctx, viamserver.SubsysName)
 		if err != nil {
 			m.logger.Warn(err)
 		}
 		m.viamServer.Update(ctx, m.cfg)
 
-		if needRestart || m.viamServerNeedsRestart {
+		if needRestart || m.viamServerNeedsRestart || m.viamAgentNeedsRestart {
 			if m.viamServer.(viamserver.RestartCheck).SafeToRestart(ctx) {
 				if err := m.viamServer.Stop(ctx); err != nil {
 					m.logger.Warn(err)
 				} else {
 					m.viamServerNeedsRestart = false
 				}
+				if m.viamAgentNeedsRestart {
```

Comment:
in cases where both agent + rdk need restart, does it make sense to check agent first, so we can restart once instead of twice? or am I thinking about this wrong

Reply:
That's what this is... we already SET this value above for agent. We're in the block checking viam-server now. So if viam server says it's safe to restart, then it doesn't matter if it NEEDS a restart, it's getting one anyway. But we're past the point where it too, would've been updated. So at this point, possibly BOTH have had new binaries installed, and agent (at least) needs to restart, so we do it once for both of them.